Sources/Markdown/Inline Nodes/Inline Containers/Link.swift

```diff
@@ -58,7 +58,7 @@ public extension Link {
             if let d = newValue, d.isEmpty {
                 _data = _data.replacingSelf(.link(destination: nil, parsedRange: nil, _data.raw.markup.copyChildren()))
             } else {
-                _data = _data.replacingSelf(.link(destination: newValue, parsedRange: nil, _data.raw.markup.copyChildren()))
+                _data = _data.replacingSelf(.link(destination: newValue, parsedRange: _data.range, _data.raw.markup.copyChildren()))
```

Comment:
Hm I worry a bit that Swift-Markdown is setting the `parsedRange` inconsistently across different markup elements. Can you explain why this change is needed? Is there another way clients could get the original range before changing the link's destination?

Reply:
> Can you explain why this change is needed?
Reason: See https://github.com/apple/swift-docc/issues/271#issuecomment-1272279547

> Is there another way clients could get the original range before changing the link's destination?

Other way IMO is to add a new property in Link so that we can differentiate []() and <doc:> in the consume site. Or we could add a new property `oldParsedRange` to store the oldRange every time we update the parsedRange.

Reply:
I see. Adding a property to `Link` that tracks what kind of link it is (i.e., a `<>` 'auto-link' or a regular `[]()` link) seems totally reasonable to me. That way we don't need to fiddle with ranges.